clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/validators/EcsCreateServerGroupDescriptionValidator.java

```diff
@@ -155,4 +158,45 @@ public void validate(List priorDescriptions, Object description, Errors errors)
       }
     }
   }
+
+  private void validateTargetGroupMappings(
+      CreateServerGroupDescription createServerGroupDescription, Errors errors) {
+    if (createServerGroupDescription.getTargetGroupMappings() != null
+        && !createServerGroupDescription.getTargetGroupMappings().isEmpty()) {
+
+      if (StringUtils.isNotEmpty(createServerGroupDescription.getTargetGroup())) {
+        // Only one of TargetGroup or TargetGroupMappings should be defined.
+        rejectValue(
+            errors,
+            "targetGroup",
+            "invalid. Specify either targetGroup or targetGroupMappings.targetGroup, not both");
```

Comment:
I think the idea is that the error code is a parseable key for strings that deck has.  i.e. deck could have a string for something like "createServerGroupDescription.targetGroup.invalid" that translates into a message, so we should create a unique error code for this validation case.

That being said, I don't think that actually happened, so it looks like in other providers you can provide an alternative string, in addition to the parseable key:
https://github.com/spinnaker/clouddriver/blob/98beae0145547edfc6f449b279409da3d87c04de/clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/validators/TitusDeployDescriptionValidator.groovy#L48

Reply:
Updated to use the spring errors.rejectValue with a defaultMessage defined: https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/validation/Errors.html#rejectValue-java.lang.String-java.lang.String-java.lang.String-